src/Parser/FeralDruid/Modules/ComboPoints/ComboPointBreakdown.js

```diff
@@ -0,0 +1,128 @@
+      }))
+      .sort((a, b) => b.spent - a.spent);
+  }
+  render() {
+    const { pointsGained, pointsSpent, pointsWasted } = this.props;
+    const generated = this.prepareGenerated(pointsGained, pointsWasted);
+    const spent = this.prepareSpent(pointsSpent);
+
+    let totalGenerated = 0;
+    let totalWasted = 0;
+    let totalSpent = 0;
+    generated.forEach(ability => {
+      totalGenerated += ability.generated;
+      totalWasted += ability.wasted;
+    });
+    spent.forEach(ability => totalSpent += ability.spent);
+    //looks wrong but totals are only for the purpose of percentage, and if nothing was wasted, then 0/1 gives correct result 0% wasted, if it's not 0 it retains its original value
+    totalGenerated = (totalGenerated === 0) ? 1 : totalGenerated;
+    totalWasted = (totalWasted === 0) ? 1 : totalWasted;
+    totalSpent = (totalSpent === 0) ? 1 : totalSpent;
+
+    return (
+      <div>
+        <table className='data-table'>
+          <thead>
+            <tr>
+              <th>Ability</th>
+              <th colSpan='2'>Combo Points generated</th>
+              <th colSpan='2'><dfn data-tip='This is the amount of points that were generated while you were already at cap.'>points wasted</dfn></th>
+            </tr>
+          </thead>
+          <tbody>
+            {generated && generated
+              .map(ability => (
+                <tr>
+                  <td style={{ width: '30%'}}>
+                    <SpellIcon id={ability.abilityId}/>{' '}
+                    <SpellLink id={ability.abilityId}/>
+                  </td>
+                  <td style={{ width: 50, paddingRight: 5, textAlign: 'right' }}>
+                    <dfn data-tip={`${formatPercentage(ability.generated / totalGenerated)} %`}>{ability.generated}</dfn>
+                  </td>
+                  <td style={{ width: '40%' }}>
+                    <div
+                      className={`performance-bar`}
+                      style={{ width: `${(ability.generated / totalGenerated) * 100}%` }}
+                    />
+                  </td>
+                  <td style={{ width: 50, paddingRight: 5, textAlign: 'right' }}>
+                    <dfn data-tip={`${formatPercentage(ability.wasted / totalWasted)} %`}>{ability.wasted}</dfn>
+                  </td>
+                  <td style={{ width: '30%' }}>
+                    <div
+                      className={`performance-bar `}
+                      style={{ width: `${(ability.wasted / totalWasted) * 100}%` }}
+                    />
+                  </td>
+                </tr>
+              ))}
+          </tbody>
+        </table>
+        <table className='data-table'>
+          <thead>
+          <tr>
+            <th>Ability</th>
+            <th colSpan='2'>Points spent</th>
+            {/* I know it shouldn't be done like this but I'm not proficient with CSS and this is the only way I could think of to align the columns with table above*/}
```

Comment:
I believe the breakdown actually comes from me, and what I meant is so that the Shards generated / Shards spent align like shown. If I omitted the last columns, it would spread out over the rest of the row and I'd rather have this code smell but so that it looks pleasing to the eye.

![image](https://user-images.githubusercontent.com/5068584/30110769-8608e28e-930b-11e7-8171-fde137d95fd1.png)

Reply:
Aligns might be wrong but this is what happens when I removed the dummy columns
![image](https://user-images.githubusercontent.com/5068584/30111181-05576514-930d-11e7-9620-60938d71c555.png)

Reply:
Yes, I used the affliction lock shard modules as a starting point since it was very similar to what I needed.  I left the comment in because I am not great with CSS either and this seemed to work better than my other ideas.